services/media/service/src/one-time-register-localization-sources.ts

```diff
@@ -0,0 +1,437 @@
+          'tvshows',
+          'tvshows_images',
+          LOCALIZATION_TVSHOW_TYPE,
+          entityId,
+          config.serviceId,
+          ownerPool,
+        );
+        payloads = batch;
+        entityId = lastEntityId;
+        break;
+      }
+      case 'SEASON': {
+        const { batch, lastEntityId } = await generateSeasonPayloads(
+          entityId,
+          config.serviceId,
+          ownerPool,
+        );
+        payloads = batch;
+        entityId = lastEntityId;
+        break;
+      }
+      case 'EPISODE': {
+        const { batch, lastEntityId } = await generateEpisodePayloads(
+          entityId,
+          config.serviceId,
+          ownerPool,
+        );
+        payloads = batch;
+        entityId = lastEntityId;
+        break;
+      }
+      case 'COLLECTION': {
+        const { batch, lastEntityId } = await generateGenericPayloads(
+          'collections',
+          'collections_images',
+          LOCALIZATION_COLLECTION_TYPE,
+          entityId,
+          config.serviceId,
+          ownerPool,
+        );
+        payloads = batch;
+        entityId = lastEntityId;
+        break;
+      }
+      case 'MOVIE_GENRE': {
+        payloads = await generateGenrePayloads(
+          'movie_genres',
+          LOCALIZATION_MOVIE_GENRE_TYPE,
+          config.serviceId,
+          ownerPool,
+        );
+        entityId = undefined;
+        break;
+      }
+      case 'TVSHOW_GENRE': {
+        payloads = await generateGenrePayloads(
+          'tvshow_genres',
+          LOCALIZATION_TVSHOW_GENRE_TYPE,
+          config.serviceId,
+          ownerPool,
+        );
+        entityId = undefined;
+        break;
+      }
+      default: {
+        continueMigration = false;
+        break;
+      }
+    }
+
+    if (typeIndex < 7) {
+      logger.log(
+        `Processing ${payloads.length} message(s) of type ${typeMapping[typeIndex].type}. Already processed: ${typeMapping[typeIndex].counter}.`,
+      );
+      typeMapping[typeIndex].counter += payloads.length;
+    }
+
+    if (payloads.length < 1000 || entityId === undefined) {
+      typeIndex++;
+      entityId = undefined;
+    }
+
+    const settings =
+      LocalizationServiceMultiTenantMessagingSettings.UpsertLocalizationSourceEntity;
+    for await (const payload of payloads) {
+      await broker.publish<UpsertLocalizationSourceEntityCommand>(
+        settings.messageType,
+        payload,
+        { auth_token: accessToken },
+        {
+          routingKey: settings.getEnvironmentRoutingKey({
+            tenantId: config.tenantId,
+            environmentId: config.environmentId,
+          }),
+        },
+      );
+      // Waiting a little for each message to be sent to make sure that all are
+      // sent before the shutdown actions are initiated, since it might take a
+      // bit of time for broker to actually send out the message.
+      await sleep(100);
```

Comment:
Maybe this can help: https://github.com/onebeyond/rascal#flow-control to pause on "busy" and continue on "ready"? Then the sleep for a correct shutdown would not be needed.

Reply:
the current broker instance is a wrapper around the `rascal` broker and it does not expose the `on` handler. Discussed internally, the current implementation is acceptable for a one-time script. For a proper implementation - a more sophisticated way to detect if all messages have been sent would have been required.